lib/logstash/outputs/elasticsearch/common_configs.rb

```diff
@@ -75,13 +75,9 @@ def self.included(mod)
       # to prevent LS from sending bulk requests to the master nodes.  So this parameter should only reference either data or client nodes in Elasticsearch.
       mod.config :hosts, :validate => :array, :default => ["127.0.0.1"]
 
-      # This plugin uses the bulk index API for improved indexing performance.
-      # This setting defines the maximum sized bulk request Logstash will make
-      # You you may want to increase this to be in line with your pipeline's batch size.
-      # If you specify a number larger than the batch size of your pipeline it will have no effect,
-      # save for the case where a filter increases the size of an inflight batch by outputting
-      # events.
-      mod.config :flush_size, :validate => :number, :default => 500
+      # DEPRECATED. We now restrict bulk sizes to a target size of 20MB. Larger events
```

Comment:
This might scare users.

Maybe say "We now automatically send the best-sized bulk request where the size is chosen to minimize network latency and memory pressure. Internally, this means we will split a bulk request into multiple requests if they would exceed 20MB in size." 

Main goal was to express the intent and also to avoid the word "restrict"